### PR TITLE
Enabled storage creation 

### PR DESCRIPTION
**Pull request by sannidhi09**
[Issue#2](https://github.com/SciGaP/emcenter-gateway/issues/2)

### CODE REVIEW

**Comment by sannidhi09** [comment]
New Storage Creation Page:

Screen Shot 2021-07-23 at 11 28 12 PM

Storage and Storage Preferences list page:

Screen Shot 2021-07-23 at 11 27 21 PM

**Comment by sannidhi09** [comment]
* Storages page path: `/storages` 
* create new storage path: `/storages/new`

File: .env [hunk 1]
```diff
@@ -1,4 +1,4 @@
-VUE_APP_CLIENT_ID=""
```
Comment (dinukadesilva) [inline .env:1]: Don't commit these changes

**Review by dinukadesilva — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (sannidhi09) [reply .env:1]: Only base path env variable should be there?

**Comment by dinukadesilva** [comment]
@sannidhi09 
1. The input fields needs to have basic validations. The hostname, port etc.
2. Storage settings menu still goes to the old storage preference page. That page may be not needed anymore and can be replaced with the storages page.
3. How about having all the storage preferences all expanded by default.
4. There's no place I can see the hostname/port of the storages. I may need to see this somewhere.

File: src/pages/storage/storage-list.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
+<template>
+  <Page title="Storage Settings" :breadcrumb-links="[]">
```
Comment (dinukadesilva) [inline src/pages/storage/storage-list.vue:2]: Map tje errors to the "Page" component so, it'll be managed accordingly.
Eg:- https://github.com/SciGaP/emcenter-gateway/blob/master/src/pages/group/group-new-page.vue#L2

**Review by dinukadesilva — CHANGES_REQUESTED** [review]

File: src/pages/storage/storage-list.vue [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                :to="{name: 'storage-preferences-new', query:{storageId: `${storage.storageId}`}}" 
+                v-slot="{ href, route, navigate}" tag="">
+                <b-button variant="link" size="sm" @click="navigate" v-b-tooltip.hover title="Create Storage Preferences">
+                  <b-icon icon="folder-plus"></b-icon>
+                </b-button>
+              </router-link>
+            </div>
+          </div>
+          <div v-if="!storageExpanded[storage.storageId]" class="w-100 p-2">
+            <table-overlay-info :rows="5" :columns="5" :data="storagePreferencesByStorageId[storage.storageId]">
```
Comment (dinukadesilva) [inline src/pages/storage/storage-list.vue:34]: There's a slot you could override to customize the empty view. 
https://github.com/apache/airavata-custos-portal/blob/master/src/lib/components/overlay/table-overlay-info.vue#L5

The following may be not suitable.
Screenshot 2021-07-27 at 14 06 50

File: src/pages/storage/storage-new.vue [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  <page :title="title" :breadcrumb-links="breadcrumbLinks" :errors="errors">
+    <template #header-right>
+      <button-overlay :show="processing">
+        <b-button variant="primary" v-on:click="onCreateClick">Create</b-button>
+      </button-overlay>
+    </template>
+    <div class="pr-3">
+      <div class="pt-3">
+        <label class="form-label">Hostname</label>
+        <b-form-input v-model="hostName" type="text" :state="hostnameValidation"/>
```
Comment (dinukadesilva) [inline src/pages/storage/storage-new.vue:11]: Please follow the style of validation done on other pages.
https://github.com/SciGaP/emcenter-gateway/blob/master/src/pages/group/group-new-page.vue#L11

File: src/pages/storage/storage-new.vue [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      </div>
+
+      <div class="pt-3">
+        <label class="form-label">Port</label>
+        <b-form-input v-model="port" type="number" :state="portValidation"/>
```
Comment (dinukadesilva) [inline src/pages/storage/storage-new.vue:16]: Let's make this a text field and show a validation error if a non-numeric value is entered. Follow existing validation style.
https://github.com/SciGaP/emcenter-gateway/blob/master/src/pages/group/group-new-page.vue#L11

**Review by dinukadesilva — CHANGES_REQUESTED** [review]

File: src/pages/storage/storage-list.vue (hunk 3 above)
Comment (sannidhi09) [reply src/pages/storage/storage-list.vue:34]: Overrides the customized empty view.

File: src/pages/storage/storage-new.vue (hunk 4 above)
Comment (sannidhi09) [reply src/pages/storage/storage-new.vue]: Added the validation and disabled the create button until all fields are validated true.

File: src/pages/storage/storage-new.vue (hunk 5 above)
Comment (sannidhi09) [reply src/pages/storage/storage-new.vue]: Changing the port field to `text` from `number` possibly creates new validation scenarios and corresponding error messages for the users to understand so I provided the placeholder(which tells the users its number field) keeping the type as `number`.

**Comment by sannidhi09** [comment]
Enabled the delete for storage preferences [Issue#3](https://github.com/SciGaP/emcenter-gateway/issues/3)

File: src/pages/storage/storage-list.vue [hunk 6]
```diff
@@ -0,0 +1,174 @@
+<template>
+  <Page title="Storage Settings" :breadcrumb-links="[]" :error=errors>
```
Comment (dinukadesilva) [inline src/pages/storage/storage-list.vue:2]: Add the breadcrumb links. Follow other pages for examples

File: src/pages/storage/storage-preference-new.vue [hunk 7]
```diff
@@ -66,13 +86,50 @@ export default {
...
+            _isFormValid &= this.isValid[this.inputFieldsList[i]];
+        }else{
+          _isFormValid &= this.isValid[this.inputFieldsList[i]];
+        }
+      }
+      return _isFormValid;
+    },
     breadcrumbLinks() {
       return [
         {to: '/storage-preferences', name: 'Storage Preferences'},
```
Comment (dinukadesilva) [inline src/pages/storage/storage-preference-new.vue:135]: Fix the link

**Review by dinukadesilva — CHANGES_REQUESTED** [review]
I get the following error when I create a storage preference

Screenshot 2021-07-29 at 09 31 09

File: src/pages/storage/storage-list.vue [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+                  </b-tr>
+                </b-thead>
+                <b-tbody>
+                  <b-tr v-for="storagePreference in storagePreferencesByStorageId[storage.storageId]" :key="storagePreference.storagePreferenceId">
+                    <b-td>{{ storagePreference.storagePreferenceId }}</b-td>
+                    <b-td>{{ storagePreference.authType }}</b-td>
+                    <b-td>{{ storagePreference.username }}</b-td>
+                    <b-td>{{ storagePreference.credentialToken }}</b-td>
+                    <b-td>
+                      <button-overlay :show="processingDelete[storagePreference.storagePreferenceId]">
```
Comment (dinukadesilva) [inline src/pages/storage/storage-list.vue:57]: `button-overlay` component is not imported. Hence no overlay for delete

**Review by dinukadesilva — CHANGES_REQUESTED** [review]

**Review by dinukadesilva — APPROVED** [review]

**Pull request merged**